backend/sublet/views.py

```diff
@@ -0,0 +1,208 @@
+        """Returns a list of Sublets that match query parameters and user ownership."""
+        # Get query parameters from request (e.g., amenities, user_owned)
+        amenities = request.query_params.getlist("amenities")
+        subletter = request.query_params.get(
+            "subletter", "false"
+        )  # Defaults to False if not specified
+        starts_before = request.query_params.get("starts_before", None)
+        starts_after = request.query_params.get("starts_after", None)
+        ends_before = request.query_params.get("ends_before", None)
+        ends_after = request.query_params.get("ends_after", None)
+        min_price = request.query_params.get("min_price", None)
+        max_price = request.query_params.get("max_price", None)
+        beds = request.query_params.get("beds", None)
+        baths = request.query_params.get("baths", None)
+
+        queryset = Sublet.objects.all().filter(expires_at__gte=timezone.now())
+
+        # Apply filters based on query parameters
+        if amenities:
+            queryset = queryset.filter(amenities__name__in=amenities)
+        if subletter.lower() == "true":
+            queryset = queryset.filter(subletter=request.user)
+        if starts_before:
+            queryset = queryset.filter(start_date__lt=starts_before)
+        if starts_after:
+            queryset = queryset.filter(start_date__gt=starts_after)
+        if ends_before:
+            queryset = queryset.filter(end_date__lt=ends_before)
+        if ends_after:
+            queryset = queryset.filter(end_date__gt=ends_after)
+        if min_price:
+            queryset = queryset.filter(min_price__gte=min_price)
+        if max_price:
+            queryset = queryset.filter(max_price__lte=max_price)
+        if beds:
+            queryset = queryset.filter(beds=beds)
+        if baths:
+            queryset = queryset.filter(baths=baths)
+
+        # Serialize and return the queryset
+        serializer = SimpleSubletSerializer(queryset, many=True)
+        return Response(serializer.data)
+
+
+class Favorites(mixins.DestroyModelMixin, mixins.CreateModelMixin, viewsets.GenericViewSet):
+    serializer_class = SubletSerializer
+    http_method_names = ["post", "delete"]
+    permission_classes = [IsAuthenticated | IsSuperUser]
+
+    def get_queryset(self):
+        user = self.request.user
+        return user.sublets_favorited
+
+    def create(self, request, *args, **kwargs):
+        sublet_id = int(self.kwargs["sublet_id"])
+        queryset = self.get_queryset()
+        if queryset.filter(id=sublet_id).exists():
+            raise exceptions.NotAcceptable("Favorite already exists")
+        sublet = get_object_or_404(Sublet, id=sublet_id)
+        self.get_queryset().add(sublet)
+        return Response(status=status.HTTP_201_CREATED)
+
+    def destroy(self, request, *args, **kwargs):
+        queryset = self.get_queryset()
+        sublet = get_object_or_404(queryset, pk=int(self.kwargs["sublet_id"]))
```

Comment:
kinda curious if this works without casting the id to int first, can we try it out?